### PR TITLE
Do not try to send email if no destination adress was provided

### PR DESCRIPTION
**Pull request by mjenczmyk**
When `toEmail` or `kindleEmail` key in config file was empty the list of emails was evaluated as `['']`, which resulted in error as the script tried to send an email to email address being empty string. This should happen no longer, as now the list parsed from config file is filtered by empty values.

I wonder if thrown exceptions shoudn't be catched somewhere in the main script.

### CODE REVIEW

File: src/utils/mail.py [hunk 1]
```diff
@@ -1,8 +1,6 @@
 #!/usr/bin/env python
-import requests
 import os
 import configparser
-import re
```
Comment (mjenczmyk) [inline src/utils/mail.py]: `requests` and `re` were unused imports.

File: src/utils/mail.py [hunk 2]
```diff
@@ -30,19 +29,22 @@ def __init__(self, cfgFilePath):
             self._smtp_port = config.get("MAIL", 'port')
             self._email_pass = config.get("MAIL", 'password')
             self._send_from = config.get("MAIL", 'email')
-            self._to_emails = config.get("MAIL", 'toEmails').split(COMMA)
-            self._kindle_emails = config.get("MAIL", 'kindleEmails').split(COMMA)
+            self._to_emails = list(filter(None, config.get("MAIL", 'toEmails').split(COMMA)))
+            self._kindle_emails = list(filter(None, config.get("MAIL", 'kindleEmails').split(COMMA)))
```
Comment (mjenczmyk) [inline src/utils/mail.py]: Here we filter emails lists by empty values.

File: src/utils/mail.py [hunk 3]
```diff
@@ -30,19 +29,22 @@ def __init__(self, cfgFilePath):
...
+            self._to_emails = list(filter(None, config.get("MAIL", 'toEmails').split(COMMA)))
+            self._kindle_emails = list(filter(None, config.get("MAIL", 'kindleEmails').split(COMMA)))
         except configparser.NoSectionError:
-            raise ValueError("ERROR: need at least one from and one or more to emails")
+            raise ValueError("ERROR: need at least one from and one or more to emails.")
 
     def send_book(self, book, to=None, subject=None, body=None):
         if not os.path.isfile(book):
-            raise
+            raise ValueError("ERROR: {} file doesn't exist.".format(book))
```
Comment (mjenczmyk) [inline src/utils/mail.py]: Raising the default exception without message seems to be ugly to me.

File: src/utils/mail.py [hunk 4]
```diff
@@ -19,6 +17,7 @@
 DEFAULT_BODY = "Enjoy!"
 DEFAULT_SUBJECT = "New free packt ebook"
 
+
 class MailBook:
```
Comment (mjenczmyk) [inline src/utils/mail.py]: PEP8 requires two empty lines before class declaration - would you like to lint that code using ex. [pylama](https://github.com/klen/pylama)?

File: src/utils/mail.py (hunk 2 above)
Comment (igbt6) [inline src/utils/mail.py]: there is a try except clause in _send_email (mail.py) which will handle this case, but yeah, i 'll add it

File: src/utils/mail.py (hunk 3 above)
Comment (igbt6) [inline src/utils/mail.py]: merged manually

File: src/utils/mail.py (hunk 1 above)
Comment (igbt6) [inline src/utils/mail.py]: good find

**Comment by igbt6** [comment]
manually merged in commit: 331aa72337e5542a9f1bfe4956d38402b3fa045e

File: src/utils/mail.py (hunk 1 above)
Comment (mjenczmyk) [inline src/utils/mail.py]: All credits go to PyCharm ;)

File: src/utils/mail.py (hunk 2 above)
Comment (mjenczmyk) [inline src/utils/mail.py]: This change is necessary for `if` in `MailBook.send_kindle` to work properly.

**Pull request closed without merging**